python/lsst/meas/extensions/piff/wcs_wrapper.py

```diff
@@ -0,0 +1,219 @@
+        d1 = np.atleast_1d(dec)
+
+        x, y = self._pix_to_sky.skyToPixelArray(r1, d1)
+
+        if np.ndim(ra) == np.ndim(dec) == 0:
+            return x[0], y[0]
+        else:
+            # Sanity checks that the inputs are the same shape.
+            assert np.ndim(ra) == np.ndim(dec)
+            assert ra.shape == dec.shape
+            return x, y
+
+    def _newOrigin(self, origin):
+        """Return a new CelestialWcsWrapper with new origin.
+
+        Parameters
+        ----------
+        origin : `galsim.PositionD`, optional
+            Origin in image coordinates.
+
+        Returns
+        -------
+        ret : `CelestialWcsWrapper`
+            Transformed WCS.
+        """
+        return CelestialWcsWrapper(self._pix_to_sky, origin=origin)
+
+
+class UVWcsWrapper(galsim.wcs.EuclideanWCS):
+    """Wrap a `lsst.afw.geom.TransformPoint2ToPoint2[2->2]` in a GalSim WCS.
+
+    Parameters
+    ----------
+    pix_to_field : `lsst.afw.geom.TransformPoint2ToPoint2[2->2]`
+        Transform to wrap.  Most likely PIXELS -> FIELD_ANGLE, but other 2D ->
+        2D transforms should be possible.
+    origin : `galsim.PositionD`, optional
+        Origin in image coordinates.
+    world_origin : `galsim.PositionD`, optional
+        Origin in world coordinates.
+    """
+    _rad_to_arcsec = 206264.80624709636
+
+    def __init__(self, pix_to_field, origin=None, world_origin=None):
+        # GalSim EuclideanWCS assumes
+        #     u = ufunc(x-x0, y-y0) + u0
+        #     v = vfunc(x-x0, y-y0) + v0
+        # where u,v are world (likely field angles), and (x, y) are pixels.
+        # GalSim also assumes that origin = x0, y0 and world_origin = u0, v0.
+        # I might have naively thought that uv(origin) == world_origin, but
+        # that appears to not be required. So we're just going to leave both
+        # free. We'll default initialize down below with both set to (0.0, 0.0)
+        # though, so ufunc is just mapping[0] and vfunc is mapping[1].
+
+        if origin is None:
+            origin = galsim._PositionD(0.0, 0.0)
+        if world_origin is None:
+            world_origin = galsim._PositionD(0.0, 0.0)
+        self._pix_to_field = pix_to_field
+        self._origin = origin
+        self._world_origin = world_origin
+        self._mapping = self._pix_to_field.getMapping()
+        self._color = None
+
+    @property
+    def origin(self):
+        """The image coordinate position to use as the origin.
+        """
+        return self._origin
+
+    @property
+    def world_origin(self):
+        """The world coordinate position to use as the origin.
+        """
+        return self._world_origin
+
+    def _xyTouv(self, x, y, color):
+        """Convert image coordinates to world coordinates.
+
+        Parameters
+        ----------
+        x, y : ndarray
+            Image coordinates.
+        color : ndarray
+            Color to use in transformation.  Unused currently.
```

Comment:
If this is not required currently, should we give a default value of `None` so they don't have to be passed in when this method is being called.